app/controllers/site_controller.rb

```diff
@@ -1,62 +1,82 @@
 class SiteController < ApplicationController
+  respond_to? :json, :html
   before_action :load_habitat
   before_action :load_global
   before_action :load_charts_data
 
-  def warmwater
-    @data = YAML.load(File.open("#{Rails.root}/lib/data/content/warmwater.yml", 'r'))
+  def index
+    @title = @habitat.title
 
-    @commitments = [
-      @aichi_targets,
-      @sdgs,
-      @data['other_targets']
-    ]
-  end
-
-  def saltmarshes
-    @data = YAML.load(File.open("#{Rails.root}/lib/data/content/saltmarshes.yml", 'r'))
-
-    @commitments = [
-      @aichi_targets,
-      @sdgs,
-      @data['other_targets']
-    ]
-  end
-
-  def mangroves
-    @data = YAML.load(File.open("#{Rails.root}/lib/data/content/mangroves.yml", 'r'))
-
-    @commitments = [
-      @aichi_targets,
-      @sdgs,
-      @data['other_targets']
-    ]
-  end
-
-  def seagrasses
-    @data = YAML.load(File.open("#{Rails.root}/lib/data/content/seagrasses.yml", 'r'))
-
-    @commitments = [
-      @aichi_targets,
-      @sdgs,
-      @data['other_targets']
-    ]
-  end
-
-  def coldcorals
-    @data = YAML.load(File.open("#{Rails.root}/lib/data/content/coldwater.yml", 'r'))
-
-    @commitments = [
-      @aichi_targets,
-      @sdgs,
-      @data['other_targets']
-    ]
+    @nav = [
+      {
+        name: 'warmwater',
+        theme: 'theme--orange',
+        title: 'Warm-water corals'
+      },
+      {
+        name: 'saltmarshes',
+        theme: 'theme--green',
+        title: 'Saltmarshes'
+      },
+      {
+        name: 'mangroves',
+        theme: 'theme--yellow',
+        title: 'Mangroves'
+      },
+      {
+        name: 'seagrasses',
+        theme: 'theme--blue',
+        title: 'Seagrasses'
+      },
+      {
+        name: 'coldcorals',
+        theme: 'theme--pink',
+        title: 'Cold-water corals'
+      }
+    ].to_json
+
+    content = YAML.load(File.open("#{Rails.root}/lib/data/content/#{@habitat.name}.yml", 'r'))
+    aichi_targets = generateImageUrls @aichi_targets
+    sdgs = generateImageUrls @sdgs
+
+    @habitatData = {
```

Comment:
We could extract this into a serialiser to improve the readability of the controller.
If not the whole hash, at least the `map` key seems the most suitable part for this.

Reply:
🤔 🤔 🤔